.circleci/config.yml

```diff
@@ -2083,43 +2077,20 @@ workflows:
           vc_product: BuildTools
           vc_version: "14.11"
           vc_year: "2017"
-      - pytorch_windows_test:
-          build_environment: pytorch-win-vs2017-14-11-cuda10-cudnn7-py3
```

Comment:
Wow, removing the test jobs here makes the library totally untested on 14.11/14.16.

Reply:
I think @malfet's contention here is that most of the signal we're getting from the Windows builds is flushing out MSVC compiler bugs. Do you think there's also runtime bugs that are likely to crop up on 14.11/14.16 but not on the most recent one?

Reply:
If I understand correctly, the binary that we are published is build using VS2019, and it seems to work on all currently supported Windows platforms, isn't it?
In that case, what is the point of testing binaries build with older versions of VisualStudio, if they are later not being used?

Reply:
@malfet No, the binaries are built with MSVC 2017 14.16 (except CUDA 9.2 binaries which use 14.11).

Reply:
I hope that we can migrate to VS 2019 but this means that we need to deprecate support for CUDA <=10.0.

Reply:
@peterjc123 is that due to the Python binary runtime requirements? Or just historically?

Reply:
> I think @malfet's contention here is that most of the signal we're getting from the Windows builds is flushing out MSVC compiler bugs. Do you think there's also runtime bugs that are likely to crop up on 14.11/14.16 but not on the most recent one?

No, we don't need to ensure PyTorch builds on every version of VC toolchain. But at least, it should work on the toolchains that we used in binary builds.

Reply:
> No, we don't need to ensure PyTorch builds on every version of VC toolchain. But at least, it should work on the toolchains that we used in binary builds.
If VS2019 results in faster/smaller binaries, why not use it for binary builds?

I also wonder if CUDA-9.2 builds on Windows can be dropped in time for next release.

Reply:
@malfet Copied from README.md
CUDA version | Newest supported VS version
-- | --
9.0 / 9.1 | Visual Studio 2017 Update 4 (15.4) (_MSC_VER <= 1911)
9.2 | Visual Studio 2017 Update 5 (15.5) (_MSC_VER <= 1912)
10.0 | Visual Studio 2017 (15.X) (_MSC_VER < 1920)
10.1 | Visual Studio 2019 (16.X) (_MSC_VER < 1930)

Old CUDA relies on old VS.

Reply:
> > No, we don't need to ensure PyTorch builds on every version of VC toolchain. But at least, it should work on the toolchains that we used in binary builds.
> > If VS2019 results in faster/smaller binaries, why not use it for binary builds?
> 
> I also wonder if CUDA-9.2 builds on Windows can be dropped in time for next release.

Well, I guess we can drop the support once CUDA 11 is out in Anaconda Cloud.

Reply:
There is already a RFC for deprecating CUDA 9.2. https://github.com/pytorch/pytorch/issues/26646 I'm okay with removing the 14.11 build/test jobs once we remove CUDA 9.2 support.

Reply:
Looks like two topics are being discussed here:
 - CUDA support matrix
 - What is currently tested in OSS CI
Leaving first question aside, I still think we don't get much extra signal from testing PyTorch build with  CUDA-10.1 and 3 different versions of Visual Studio. And this is what I'm proposing to  do with this PR: change build/test matrix to cover different compiler builds but test only one out of 3 configurations, since, to the best of my knowledge, there are very few examples of PyTorch runtime behavior is drastically different(i.e. test fails in one config but works in another) between binaries compiled using different version of compiler.

Reply:
@malfet Yes, maybe you are right. But actually the build doesn't cover all the codepaths. I strongly suggest we should still test building the c++ extensions at least.

Reply:
Update `binary_windows_builds` to use VS2019